Source/WebCore/editing/Editor.cpp

```diff
@@ -3692,7 +3692,7 @@ bool Editor::findString(const String& target, FindOptions options)
         resultRange = rangeOfString(target, selection.firstRange(), options);
     }
 
-    if (!resultRange)
+    if (!resultRange || resultRange->startContainer().renderer()->style().effectiveInert())
```

Comment:
This isn't quite right. `rangeOfString` needs to be skipping inert contents instead of making this function return false at the last minute like this. Specifically, we probably need to make `TextIterator` skip inert content. See how it checks `renderer->isSkippedContentRoot()` to skip contents for `content-visibility: hidden`.

Reply:
OK, a lot of test failures already at https://ews-build.webkit.org/#/builders/1/builds/40991 and https://ews-build.webkit.org/#/builders/36/builds/37244 — and looking at them and thinking about it further makes realize: We don’t actually want/need to be changing `TextIterator::advance` to be ignoring (skipping over) inert nodes in all cases, right?

I mean, that’s not what the spec defines. The relevant spec requirement here is instead actually limited just to the case of find-in-page and ignoring inert nodes in just that case.

So doesn’t the behavior for that specific find-in-page requirement actually need to be implemented in `Editor::findString` after all?

Reply:
> `rangeOfString` needs to be skipping inert contents instead of making this function return false at the last minute like this.

OK, after stepping back and taking some time to just think about it some more, I realize that maybe what you were getting at was something more like, passing a new _option_ to `rangeOfString` that would make it optionally skip inert nodes in the cases where that’s what the calling code needs. (Which right now is just this find-in-page case, but it’s imaginable that more cases could come along later.)

Anyway, I’ll take shot at reworking the patch along those lines.